test/src/helpers/use-document-head.test.tsx

```diff
@@ -0,0 +1,255 @@
+                    Calculus textbook and using our accompanying online resources.`,
+                locale: 'en'
+            },
+            title: 'Calculus Volume 1',
+            bookSubjects: [
+                {
+                    id: 177,
+                    meta: {
+                        type: 'books.BookSubjects'
+                    },
+                    subjectName: 'Math',
+                    subjectPageContent:
+                        'About this much text. Lorem ipsum dolor sit amet, consectetur ...',
+                    subjectSearchDescription: 'This is math search meta'
+                }
+            ],
+            bookCategories: [
+                {
+                    id: 18,
+                    meta: {
+                        type: 'books.BookCategories'
+                    },
+                    subjectName: 'Math',
+                    subjectCategory: 'Calculus'
+                }
+            ],
+            description: 'some HTML description',
+            k12bookSubjects: [],
+            bookStudentResources: [],
+            bookFacultyResources: [],
+            publishDate: '2016-03-30',
+            authors: [],
+            printIsbn13: '978-1-938168-02-4',
+            slug: 'books/calculus-volume-1'
+        };
+        const Component = () => {
+            React.useEffect(
+                () => setPageTitleAndDescriptionFromBookData(data),
+                []
+            );
+
+            return null;
+        };
+
+        render(<Component />);
+        await waitFor(() =>
+            expect(document.title).toBe(`${data.meta.seoTitle} - OpenStax`)
+        );
+
+        expect(getPageDescription()).toBe(data.meta.searchDescription);
+        expect(console.warn).not.toHaveBeenCalled();
+    });
+});
+
+// The above created a <title> in <head> that is used below
+
+describe('useCanonicalLink', () => {
+    it('sets nothing when arg is false', () => {
+        const Component = () => {
+            useCanonicalLink(false);
+            return null;
+        };
+
+        render(
+            <MemoryRouter initialEntries={['/foo/dir']}>
+                <Component />
+            </MemoryRouter>
+        );
+        expect(document.head.querySelector('link')).toBeNull();
+    });
+    it('sets link with default args', () => {
+        const canonicalPath = '/canonical/path';
+        const Component = () => {
+            useCanonicalLink();
+            return null;
+        };
+
+        render(
+            <MemoryRouter initialEntries={[canonicalPath]}>
+                <Component />
+            </MemoryRouter>
+        );
+        expect(
+            document.head.querySelector('link')?.getAttribute('href')
+        ).toContain(canonicalPath);
+    });
+    it('handles fails gracefully if there is no title', () => {
+        const titleEl = document.head.querySelector('title');
+
+        document.head.removeChild(titleEl as Node);
```

Comment:
I see the comment above... maybe a guard here though so it doesn't error if the test or describe block is focused?